From 1e8ee0a4dd473674968f64e0cd445ad0c12fa8ff Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Wed, 22 Nov 2023 06:51:59 +0100
Subject: [PATCH 31/55] libubox: use CLOCK_BOOTTIME

Changes uloop to use CLOCK_BOOTTIME instead of CLOCK_MONOTONIC, ensuring timers continue counting during system suspend.

Signed-off-by: John Crispin <john@phrozen.org>
---
 .../libs/libubox/patches/001-CLOCK_BOOTTIME.patch   | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 package/libs/libubox/patches/001-CLOCK_BOOTTIME.patch

diff --git a/package/libs/libubox/patches/001-CLOCK_BOOTTIME.patch b/package/libs/libubox/patches/001-CLOCK_BOOTTIME.patch
new file mode 100644
index 0000000000..5e78d5ade9
--- /dev/null
+++ b/package/libs/libubox/patches/001-CLOCK_BOOTTIME.patch
@@ -0,0 +1,13 @@
+Index: libubox-2023-05-23-82fa6480/uloop.c
+===================================================================
+--- libubox-2023-05-23-82fa6480.orig/uloop.c
++++ libubox-2023-05-23-82fa6480/uloop.c
+@@ -293,7 +293,7 @@ static void uloop_gettime(struct timeval
+ {
+ 	struct timespec ts;
+ 
+-	clock_gettime(CLOCK_MONOTONIC, &ts);
++	clock_gettime(CLOCK_BOOTTIME, &ts);
+ 	tv->tv_sec = ts.tv_sec;
+ 	tv->tv_usec = ts.tv_nsec / 1000;
+ }
-- 
2.34.1

